Allow uppercase chars in Shared Folder name

61f7b7437262085c97876892ddf9c0d8 opened the pull request:
The username validator is too restrictive, removed completely.

NethServer/dev#5247

61f7b7437262085c97876892ddf9c0d8 committed:
Allow uppercase chars in Shared Folder name

The username validator is too restrictive, replaced with a new validator ibay-name.

gsanchietti commented:
In-progress...

gsanchietti reviewed (changes requested):
This PR is still in progress.

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_SharedFolder.php:
```diff
@@ -41,3 +41,4 @@
 $L['migrate-vhost_Message'] = 'This shared folder will be deleted and all data will be moved inside the virtual host. Migrate the shared folder to virtual host `${0}`?';
 $L['Migrate_label'] = 'Migrate';
 $L['__ANY__'] = 'default';
+$L['valid_platform,ibay-validator,ibay-name,3'] = 'The following characters are not allowed \ / : * ? " < > |';
```
The label key must be fixed: ``valid_platform,ibay-create,ibay-name,3`` (``validator`` becomes ``create``).

gsanchietti commented on createlinks:
```diff
@@ -56,6 +56,7 @@ event_actions('nethserver-ibays-update', qw(
 validator_actions('ibay-create', qw(
                   uniq-account 20
                   ibay-length 30
+				  ibay-name 20
```
Please correctly indent this line.

gsanchietti commented on root/etc/e-smith/validators/actions/ibay-name:
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/perl
+#
+# Copyright (C) 2012 Nethesis S.r.l.
```
Change the year to 2017.

61f7b7437262085c97876892ddf9c0d8 committed:
Square brackets added to not allowed characters

Shared folder names with square brackets make same troubles in sbm.conf file.
Square brackets are added to not allowed characters.
